### PR TITLE
Handle lingering_data

### PR DESCRIPTION
**Pull request by tony612**
See https://github.com/ninenines/cowlib/pull/82

### CODE REVIEW

**Comment by essen** [comment]
Thanks, but you're missing a test for me to reproduce this.

**Comment by tony612** [comment]
I added the test, but I have to change `dep_cowlib` in Makefile to let the test pass because it depends on the cowlib PR.

**Comment by essen** [comment]
Thanks! I'm taking a look tomorrow.

File: test/rfc7540_SUITE.erl [hunk 1]
```diff
@@ -3116,6 +3117,59 @@ data_reject_overflow_stream(Config0) ->
...
+		% Make sure server send RST_STREAM
+		timer:sleep(100),
+		ok = gen_tcp:send(Socket, [
+			cow_http2:data(1, fin, <<0:1000/unit:8>>)
+		]),
+		{ok, << SkipLen:24, 1:8, _:8, 1:32 >>} = gen_tcp:recv(Socket, 9, 1000),
+		% Skip the header
+		{ok, _} = gen_tcp:recv(Socket, SkipLen, 1000),
+		% @todo: A WINDOW_UPDATE frame is expected to be received, but cowboy_stream_h
+		% 	doesn't send it when only first 1000 bytes of body is read
```
Comment (essen) [inline test/rfc7540_SUITE.erl]: Why the todo?

**Comment by essen** [comment]
I removed the todo, did some tweaks and merged. Thanks!

**Pull request closed without merging**